### PR TITLE
Update JavaScript build process to work better with Gutenberg

### PR DESCRIPTION
**Pull request by ryanwelcher**
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my change.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "@10up/eslint-config",
```
Comment (nicholasio) [inline .eslintrc]: @ryanwelcher Can you try using the new eslint config? https://github.com/10up/eslint-config/pull/17 It's still being worked on but you can already install it with `npm install @10up/eslint-config@next` there's a Gutenberg specific config

Any feedback is appreciated.

File: src/js/admin.js [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
-	const $userFieldWrapper = $userField.closest( 'tr' );
-	const [$passwordFieldTitle] = document.getElementById( 'classifai-settings-watson_password' ).closest( 'tr' ).getElementsByTagName( 'label' );
+	const $userFieldWrapper = $userField.closest('tr'); // eslint-disable-line
+	const [$passwordFieldTitle] = document // eslint-disable-line
+		.getElementById('classifai-settings-watson_password')
+		.closest('tr')
+		.getElementsByTagName('label');
 
-	if ( null === $toggler ) return;
+	if ($toggler === null) return;
```
Comment (nicholasio) [inline src/js/admin.js:14]: This if check is actually not needed. Line 6 already takes care of it.

Comment (nicholasio) [inline src/js/admin.js]: This if check is not needed. See Line 6

File: src/js/admin.js [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
+	const $toggler = document.getElementById('classifai-waston-cred-toggle');
+	const $userField = document.getElementById('classifai-settings-watson_username');
 
-	if ( null === $toggler || null === $userField ) return;
+	if ($toggler === null || $userField === null) return;
 
-	const $userFieldWrapper = $userField.closest( 'tr' );
-	const [$passwordFieldTitle] = document.getElementById( 'classifai-settings-watson_password' ).closest( 'tr' ).getElementsByTagName( 'label' );
+	const $userFieldWrapper = $userField.closest('tr'); // eslint-disable-line
+	const [$passwordFieldTitle] = document // eslint-disable-line
```
Comment (nicholasio) [inline src/js/admin.js]: If you remove Line 14 you don't need to ignore these two lines.

File: src/js/editor.js [hunk 4]
```diff
@@ -1,23 +1,29 @@
-
-const { subscribe, select, dispatch } = wp.data;
+import { subscribe, select, dispatch } from '@wordpress/data'; // eslint-disable-line
```
Comment (nicholasio) [inline src/js/editor.js]: I've updated the eslint config to account for "@wordpress" imports.

File: .eslintrc [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+	"extends": "@10up/eslint-config/gutenberg",
```
Comment (nicholasio) [inline .eslintrc]: Change to `@10up/eslint-config/wordpress`

File: package.json [hunk 6]
```diff
@@ -30,19 +30,34 @@
     "role": "developer"
   },
   "devDependencies": {
-    "@10up/eslint-config": "^1.0.9",
+    "@10up/eslint-config": "^2.0.0-next",
```
Comment (nicholasio) [inline package.json]: Update to `2.0.2-next`

**Review by nicholasio — CHANGES_REQUESTED** [review]
@ryanwelcher I updated the eslint config to address some of the issues here. I've also renamed the config name from `10up/eslint-config/gutenberg` to `10up/eslint-config/wordpress` as it makes more sense.

**Comment by helen** [comment]
Could I get a summary here about why we're using the 10up config instead of `wp-scripts`? I am not strongly opinionated except that I am not sure why there is a WP-specific config inside 10up's package when a WP one already exists.

**Comment by jeffpaul** [comment]
@helen @ryanwelcher @nicholasio @dinhtungdu @dkotter given that we're adding a Gutenberg integration as part of #228 do we want to revisit this PR and bring into the 1.6.0 release if it'll help in relation to the work in 228?

**Comment by dinhtungdu** [comment]
@jeffpaul I prefer to keep this one for 1.7 because it's not a blocker for 228.

**Comment by ryanwelcher** [comment]
@dinhtungdu @jeffpaul based on the changes to our tooling since this PR was created, I would probably recommend removing the custom webpack config in favor of using everything provided by [10up-toolkit](https://github.com/10up/10up-toolkit). 

This is the [standard approach for new projects ](https://github.com/10up/wp-scaffold/blob/trunk/themes/10up-theme/package.json)and provides all of the linting tools we'd need here. 

Additionally, [we can leverage the package.json file](https://github.com/10up/wp-scaffold/blob/trunk/themes/10up-theme/package.json#L23) to manage customizations which means that we won't need a custom webpack config file

**Comment by helen** [comment]
I'm closing this out for now, it's so out of date we are better off starting over again. I am not yet sure about 10up-toolkit for OSS WP plugins, as there are differences between something that happens to be openly available but still geared toward client work, and what we want to represent in a community project. We need to take a look at this across 10up plugins and then go from there :)

**Pull request closed without merging**